sympy/stats/frv.py

```diff
@@ -306,6 +310,18 @@ def compute_moment_generating_function(self, expr):
         return Lambda(t, sum(exp(k * t) * v for k, v in d.items()))
 
     def compute_expectation(self, expr, rvs=None, **kwargs):
+        from sympy.stats.frv_types import DieDistribution, BinomialDistribution, \
+                                            HypergeometricDistribution
+        rands = random_symbols(expr)
+        for rv in rands:
+            dist = rv.pspace.distribution
+            cond1 = isinstance(dist, DieDistribution) and dist.sides.has(Symbol)
+            cond2 = isinstance(dist, BinomialDistribution) and dist.n.has(Symbol)
+            cond3 = isinstance(dist, HypergeometricDistribution) and \
+                    (dist.N.has(Symbol) or dist.n.has(Symbol) or dist.m.has(Symbol))
```

Comment:
Rather than explicitly listing the distributions here it would be better to add some attribute to the distributions that can be used to query whether the dimensions are symbolic

Reply:
Definitely a bad idea to check individual distributions.